lutris/sysoptions.py

```diff
@@ -82,13 +91,38 @@ def get_optirun_choices():
     return choices
 
 
-def get_gpu_vendor_cmd(is_nvidia):
-    """Run glxinfo command to get vendor based on certain conditions"""
-    if is_nvidia:
-        return "__GLX_VENDOR_LIBRARY_NAME=nvidia glxinfo | grep -i opengl | grep -i vendor"
+def get_vulkan_gpus(env):
+    gpu = get_vulkan_gpu(env)
     if USE_DRI_PRIME:
-        return "DRI_PRIME=1 glxinfo | grep -i opengl | grep -i vendor"
-    return "glxinfo | grep -i opengl | grep -i vendor"
+        prime_gpu = get_vulkan_gpu("DRI_PRIME=1 " + env)
+        if prime_gpu != gpu:
+            gpu += f" (Discrete GPU: {prime_gpu})"
+    return gpu
+
+
+def get_vulkan_gpu(env):
+    infocmd = get_vulkan_gpu_cmd(env)
+    with subprocess.Popen(infocmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as infoget:
+        result = infoget.communicate()[0].decode("utf-8").strip()
+    if "Failed to detect any valid GPUs" in result or "ERROR: [Loader Message]" in result:
+        return "No GPUs"
+    return result
+
+
+
+def get_vulkan_gpu_cmd(env):
+    """Run vulkaninfo (or glxinfo as fallback) command to get vendor based on certain conditions"""
```

Comment:
This is not how Lutris usually handles environments.

'env' would normally be a dict, and passed as the 'env' parameter of Popen, and os.environ.copy() to obtain the parent enviroment first.

If we must prefix some shell stuff - I don't really like it - then at least the parameter name should not be 'env'. It is asking for a dict to be passed as per convention.

Reply:
Got it, I'll sort this out.

Reply:
Swapped it to use the env parameter.